source/CellMeasurer/CellMeasurer.js

```diff
@@ -0,0 +1,222 @@
+     * If this property is not specified the document body will be used.
+     */
+    container: React.PropTypes.oneOfType([
+      React.PropTypes.func,
+      React.PropTypes.node
+    ]),
+
+    /**
+     * Assign a fixed :height in order to measure dynamic text :width only.
+     */
+    height: PropTypes.number,
+
+    /**
+     * Number of rows in grid.
+     */
+    rowCount: PropTypes.number.isRequired,
+
+    /**
+     * Assign a fixed :width in order to measure dynamic text :height only.
+     */
+    width: PropTypes.number
+  };
+
+  constructor (props, state) {
+    super(props, state)
+
+    this._cachedColumnWidths = {}
+    this._cachedRowHeights = {}
+
+    this.getColumnWidth = this.getColumnWidth.bind(this)
+    this.getRowHeight = this.getRowHeight.bind(this)
+    this.resetMeasurements = this.resetMeasurements.bind(this)
+  }
+
+  getColumnWidth ({ index }) {
+    if (this._cachedColumnWidths[index]) {
+      return this._cachedColumnWidths[index]
+    }
+
+    const { rowCount } = this.props
+
+    let maxWidth = 0
+
+    for (var rowIndex = 0; rowIndex < rowCount; rowIndex++) {
+      let { width } = this._measureCell({
+        columnIndex: index,
+        rowIndex
+      })
+
+      maxWidth = Math.max(maxWidth, width)
+    }
+
+    this._cachedColumnWidths[index] = maxWidth
+
+    return maxWidth
+  }
+
+  getRowHeight ({ index }) {
+    if (this._cachedRowHeights[index]) {
+      return this._cachedRowHeights[index]
+    }
+
+    const { rowCount } = this.props
+
+    let maxHeight = 0
+
+    for (var columnIndex = 0; columnIndex < rowCount; columnIndex++) {
+      let { height } = this._measureCell({
+        columnIndex,
+        rowIndex: index
+      })
+
+      maxHeight = Math.max(maxHeight, height)
+    }
+
+    this._cachedRowHeights[index] = maxHeight
+
+    return maxHeight
+  }
+
+  resetMeasurements () {
+    this._cachedColumnWidths = {}
+    this._cachedRowHeights = {}
+  }
+
+  componentDidMount () {
+    this._renderAndMount()
+  }
+
+  componentWillReceiveProps (nextProps) {
+    this._updateDivDimensions(nextProps)
+  }
+
+  componentWillUnmount () {
+    this._unrenderAndDestroy()
+  }
+
+  render () {
+    const { children } = this.props
+
+    return children({
+      getColumnWidth: this.getColumnWidth,
+      getRowHeight: this.getRowHeight,
+      resetMeasurements: this.resetMeasurements
+    })
+  }
+
+  _getContainerNode (props) {
+    const { container } = props
+
+    if (container) {
+      return ReactDOM.findDOMNode(
+        typeof container === 'function'
+          ? container()
+          : container
+      )
+    } else {
+      const node = ReactDOM.findDOMNode(this)
+
+      return node.ownerDocument.body
+    }
+  }
+
+  _measureCell ({
+    columnIndex,
+    rowIndex
+  }) {
+    const { cellRenderer } = this.props
+
+    const rendered = cellRenderer({
+      columnIndex,
+      rowIndex
+    })
+
+    this._div.innerHTML = ReactDOMServer.renderToString(rendered)
```

Comment:
I also have big reservations about this line. I actually tried a couple of render approaches like you mention- but React threw an error about functions (eg. `getRowHeight`) having side effects that trigger rendering.

I think the main problem is that I want the width/height getter functions to be synchronous, not callback based like yours. Because (a) I don't want to force everyone to use the `InfiniteLoader` for the callback and (b) I think the callback approach you're using might cause a lot of `forceUpdates`.

I'd love to improve this bit though. Let's talk more about ideas?